tokio/src/runtime/basic_scheduler.rs

```diff
@@ -472,6 +482,8 @@ impl SchedulerPriv {
 impl Schedule for SchedulerPriv {
     fn bind(&self, task: &Task<Self>) {
         unsafe {
+            // safety: tasks are only bound to a basic scheduler from the local
+            // thread it runs on, so it's safe to call this.
```

Comment:
How are we guaranteed to be on the same thread? Is `SchedulerPriv` not `Sync`/`Send`?

Reply:
`bind` is only called when we poll a task for the first time. this only happens when the basic scheduler is running, and all the tasks are polled from one thread.

Reply:
Hopefully this is clearer in https://github.com/tokio-rs/tokio/pull/1892/commits/03357f1b78744fd4bf3f963859f8573c22b25d6e?